table/block_based/block_based_table_builder.cc

```diff
@@ -460,8 +474,7 @@ struct BlockBasedTableBuilder::Rep {
             _moptions.prefix_extractor != nullptr));
     if (table_options.verify_compression) {
       for (uint32_t i = 0; i < compression_opts.parallel_threads; i++) {
-        verify_ctxs[i].reset(new UncompressionContext(
-            UncompressionContext::NoCache(), compression_type));
+        verify_ctxs[i].reset(new UncompressionContext(compression_type));
```

Comment:
BTW, I find `NoCache` here (introduced in #3838) will lead to a null uncompression context when using ZSTD during compression verification.

I'm not sure why the `NoCache` is used here, since when compression verification is enabled, non-null ZSTD uncompression context is a must. I also found  #5496 saying "Rocksdb should not verify compression if dictionary is not set instead of crashing", but seems ZSTD uncompression can be done both with or without dictionary. As a result, I think it's a bug here.

Please correct me if I'm wrong.

Reply:
I wonder whether @ajkr knows the answer.